app/cli/cmd/root.go

```diff
@@ -211,3 +232,53 @@ func loadControlplaneAuthToken(cmd *cobra.Command) (string, error) {
 	// loaded from config file, previously stored via "auth login"
 	return viper.GetString(confOptions.authToken.viperKey), nil
 }
+
+// parseToken the token and return the type of token. At the moment in Chainloop we have 3 types of tokens:
+// 1. Robot account token
+// 2. User account token
+// 3. API token
+// Each one of them have an associated audience claim that we use to identify the type of token. If the token is not
+// present, nor we cannot match it with one of the expected audience, return nil.
+func parseToken(token string) (*ParsedToken, error) {
+	if token == "" {
+		return &ParsedToken{}, nil
+	}
+
+	// Create a parser without claims validation
+	parser := jwt.NewParser(jwt.WithoutClaimsValidation())
+
+	// Parse the token without verification
+	parsedToken, _, err := parser.ParseUnverified(token, jwt.MapClaims{})
+	if err != nil {
+		return nil, err
+	}
+
+	// Extract claims
+	claims := parsedToken.Claims.(jwt.MapClaims)
+
+	// Get the audience claim
+	if val, ok := claims["aud"]; ok && val != nil {
+		// Chainloop tokens have only one audience in an array
+		aud, ok := val.([]interface{})
+		if !ok {
+			return nil, nil
+		}
+		if len(aud) == 0 {
+			return nil, nil
+		}
+
+		switch aud[0].(string) {
+		case apiTokenAudience:
+			return &ParsedToken{Type: "api-token"}, nil
```

Comment:
my bad, shouldn't we add the identifier of the token? That way we can know how much for example a single token is being used in an org?

Reply:
and I mean the identifier of both the API_TOKEN and the RobotAccount

Reply:
Do you see it as the `ID` on the `ParsedToken` or a new field `TokenID`?

Reply:
mm, I was thinking that maybe using the current ID in the ParsedToken? Why do you think we should do it as TokenID?

Reply:
No no, to have different fields to filter by on the UI but we can merge them into `ID` since we have the type of token to know if it's a user.

Reply:
ahh, I see, no idea tbh. You probably know more.